external_server/logs.py

```diff
@@ -0,0 +1,148 @@
+from typing import Optional, Type
+import abc
+import logging.handlers
+import os
+
+from external_server.models.exceptions import (  # type: ignore
+    ConnectSequenceFailure,
+    StatusTimeout,
+    NoPublishedMessage,
+    CommandResponseTimeout,
+    SessionTimeout,
+    UnexpectedMQTTDisconnect,
+)
+
+
+LOGGER_NAME = "external_server"
+_DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+_log_level_by_verbosity = {False: logging.WARNING, True: logging.DEBUG}
+
+
+class _Logger(abc.ABC):
```

Comment:
Add docstring to all these loggers. Explain their purposes simply

Reply:
Done